cEP-0020.md

```diff
@@ -0,0 +1,529 @@
+to contribute and engage with coala and its community.
+
+## Gamification Introduction
+
+There will be four game design elements to help newcomers:
+Activities, Points, Badges and Levels. For each of these
+elements, we will have a set of rules to describe the
+operation of the gamified environment.
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create a bug report
+- Create newcomer difficulty issue
+- Create a bear proposal issue
+- Create a documentation issue
+- Create low difficulty issue
+- Create medium difficulty issue
+- Create pull request that fixes a bug
+- Create pull request for bear proposal
+- Create pull request for a documentation issue
+- Create pull request for newcomer issue
+- Create pull request for low difficulty issue
+- Create pull request for medium difficulty issue
+- Gitmate errors on pull request
+- CI status of the pull request
+- Review comments on pull request
+- Number of questions asked in `coala/coala` room
+- Number of answers given in `coala/coala` room
+- Number of questions asked in `coala-bears` room
+- Number of  answers given in `coala-bears` room
+- Number of questions asked in `coala/corobo` room
+- Number of answers given in `coala/corobo` room
+- Number of questions asked in `coala/community` room
+- Number of answers given in `coala/community` room
+- Number of messages in coala other rooms
+- Make use of `corobo` on Gitter
+- Introduce `.coafile` in user own repository
+- Introduce `.coafile` in an organisation with 50 or more stars
+  for its repositories
+- Introduce CI in the users own repository to run coala using
+  Travis CI
+- Setup the Circle CI to the fork of coala repository
+- Setup Travis CI to the fork of coala repository
+- Setup Appveyor CI to the fork of coala repository
+
+### Points
+
+Points are a simple tool to reward positive contributions
+to coala, and they provide an excellent base upon which to
+build a gamified design. Now let's assign points to some
+of the tracked activities mentioned above:
+
+- Create Issue
+  - 20 points for type/bug
+  - 20 points for bear proposal
+  - 10 points for difficulty newcomer
+  - 20 points for difficulty low
+  - 30 points for difficulty medium
+- Merged Pull Request
+  - If there are no Gitmate errors
+    - 20 points for bug fixes
+    - 20 points for writing a bear
+    - 10 points for difficulty newcomer
+    - 20 points for difficulty low
+    - 30 points for difficulty medium
+  - If Gitmate errors are b/w 1 to 5
+    - 15 points for bug fixes
+    - 15 points for writing a bear
+    - 7 points for difficulty newcomer
+    - 15 points for difficulty low
+    - 20 points for difficulty medium
+  - If Gitmate errors are b/w 5 to 10
+    - 10 points for bug fixes
+    - 10 points for writing a bear
+    - 5 points for difficulty newcomer
+    - 10 points for difficulty low
+    - 15 points for difficulty medium
+  - If Gitmate errors are b/w 10 and beyond
+    - No points are given
```

Comment:
Tbh I think merges are actually worth way more points. Filing an issue is much quicker than solving it^^

Reply:
Unfortunately we havent done any testing of these points with sample people.  We definitely need to, before the end of the project, in order to validate the points allocated, and tweak accordingly.

@sks444 , can you create a milestone task in phase 2 to build a dataset for 10 of the 2018 gsoc students as of Jan 2018, and a dataset as of March 2018 & a task for phase 3 to validate the points allocated against those two datasets.  We need to verify that 2018 gsoc students as of Jan 2018, and as of March 2018, would have been given points & levels which feel sensible, and tweak the configuration to as required to get a better outcome.

Filing a half-decent non-duplicate bug report (not a feature) is usually much harder than solving it, unless it is of difficulty/medium or higher.  issues do need to be incentivised, because without issues there are no tasks to allocate, and we have way too many people turning up at a hackathon and doing a pretty crappy MR and then disappearing, which is a net negative on the community, costing us review time without getting a commit for the effort.  If they instead used coala, and found a bug, they can disappear after the hackathon and we have a net positive value.

Anyway, there is definitely room to improve these points allocations, and we'll be able to (and will need to) adjust these after go-live to try to find the best balance.  I've created a few comments above to have the allocations revised.

specifically, regarding a MR with lots of gitmate errors, it should still have some a low points allocated if it is merged.
Otherwise we encourage people to give up after 10 gitmate errors, or close their PR and create a new one hoping they will get more points on the new one.

And this raises a problem that has somehow escaped this cEP until now. negative points. (we have discussed it, but they dont appear in the cEP yet)
Creation of duplicate PRs and duplicate issues is a great example of an activity which must be actively decentivised, otherwise the points system described here will fall apart.